course4/homework-string/string.cc

```diff
@@ -1,9 +1,145 @@
 #include "string.hh"
+#include <cstring>
+#include <memory>
 
 namespace course {
 
 string::string()
 {
+    clear();
 }
 
-}
\ No newline at end of file
+string::string(const char * c)
+{
+    copyChars(c);
+}
+
+string::string(const string& source)
+{
+    copyChars(source.chars.get());
+}
+
+string::string(string&& source)
+{
+    stealChars(std::move(source));
+}
+
+bool string::empty() const
+{
+    return !mSize;
+}
+
+int string::size() const
+{
+    return mSize;
+}
+
+int string::length() const
+{
+    return mSize;
+}
+
+void string::clear()
+{
+    mSize = 0;
+    chars = std::make_unique<char[]>(1);
```

Comment:
It would be better to avoid this allocation. Seems wasteful to perform expensive operation just for one byte; and a one that never changes.